### ISSUE TITLE
Add support for x64 program

### ISSUE THREAD

**Issue by RadWolfie** [Cxbx-Reloaded/XbSymbolDatabase#25]
I noticed some warning outputs for x64 build by accident from AppVeyor's build.
```
XbSymbolDatabase.c(284): warning C4312: 'type cast': conversion from 'uint32_t' to 'uint32_t *' of greater size [C:\projects\cxbx-reloaded\import\XbSymbolDatabase\xbSymbolDatabase.vcxproj]
...
XbSymbolDatabase.c(655): warning C4312: 'type cast': conversion from 'unsigned int' to 'unsigned int *' of greater size [C:\projects\cxbx-reloaded\import\XbSymbolDatabase\xbSymbolDatabase.vcxproj]
...
XbSymbolDatabase.c(1104): warning C4267: 'function': conversion from 'size_t' to 'const unsigned int', possible loss of data [C:\projects\cxbx-reloaded\import\XbSymbolDatabase\xbSymbolDatabase.vcxproj]
```

If we need add support for such as IDA Pro's plugin feature. It's something we'll have to do to fix.

**Comment by PatrickvL** [comment]
Xbox code will at most use 32 bit memory address, so any address or offset can be stored with a 32 bit number. But 64 bit plugins must be prepared to access actual buffer addresses above 32 bit. So that should use either void pointer, or uintptr_t

**Comment by JayFoxRox** [comment]
In my opinion, most of the code should still only use 32-bit addresses. The re-mapping from Xbox to host space should occur in separate address translation / virtual memory read functions.

**Comment by RadWolfie** [comment]
Much as I would like to stay specific 32-bit addresses. Except there would not be a good support for 64-bit's 4GB higher range address. The scanning process has to be able support it. What we could do is hard-coded found xbox's 32-bit address to callback's register function.

**Comment by JayFoxRox** [comment]
I merely meant that the address calculation for Xbox addresses should remain 32-bit. I'm fine with translating addresses for buffer lookups to some other type: `void* translate_xbox_address(uint32_t xbox_virtual_address[, size_t byte_count])` (as final step).
There could also be memory accessor functions like `uint32_t read_xbox_u32(uint32_t xbox_virtual_address)`.

This translation function could be supplied as a callback (to make it the users responsibility), or be part of XbSymbolDatabase (assuming that the memory layout is non-paged and XBE virtual memory image is entirely loaded into memory with ImageSize bytes, at whatever base the user wants).
I prefer it be part of XbSymbolDatabase for simplicity.

**Comment by RadWolfie** [comment]
Okay, I understand what you meant now. 😉 We're on the same page of understanding.

> (assuming that the memory layout is non-paged and XBE virtual memory image is entirely loaded into memory with ImageSize bytes, at whatever base the user wants)

That's the plan since @PatrickvL had request this type of support.

**Comment by RadWolfie** [comment]
Waiting on pull request #47 to be merge before create another pull request to finalize support x64 programs.
